src/parse_collection.cpp

```diff
@@ -86,28 +88,29 @@ std::function<std::string(std::string &&)> term_processor(std::optional<std::str
 {
     if (not type) {
         return [](std::string &&term) -> std::string {
-            boost::algorithm::to_lower(term); 
-            return term;
+            boost::algorithm::to_lower(term);
+            return std::move(term);
```

Comment:
You mean `std::forward` here?

Reply:
I don't think forward makes sense here because this is not a forwarding reference (no template) but simply an rvalue reference. What I'm now thinking is whether I need that nice or not. We could simplify this by passing by value actually. The idea was to force to move a value. If we pass by value, we could be copying by mistake if we don't move at call site.

What do you think?

Reply:
Yes, you can pass by value and then move. In this case, when you call this function, you have to make sure you move when calling the lambda to avoid an extra copy though.

If you pass by forwarding reference the risk of using a move when returning is that you could pass an lvalue reference, which then gets moved (resulting in an extra copy). That’s why I wanted to use forward here.

Reply:
Actually this is fine because it is not templated.

Reply:
Yeah, I was gonna say.